teensytoany/teensytoany.py

```diff
@@ -260,11 +265,18 @@ def _generate_firmware_directory(*, mcu):
         return firmware_dir
 
     @staticmethod
-    def download_firmware(*, mcu, version, timeout=2):
-        firmware_filename = TeensyToAny._generate_firmware_filename(mcu=mcu, version=version)
+    def download_firmware(*, mcu, version, variant: str=None, timeout=2):
+        firmware_filename = TeensyToAny._generate_firmware_filename(
+            mcu=mcu,
+            version=version,
+            variant=variant
+        )
 
         import requests  # pylint: disable=import-outside-toplevel
-        file_url = f"https://github.com/ramonaoptics/teensy-to-any/releases/download/{version}/firmware_{mcu.lower()}.hex"  # noqa # pylint: disable=line-too-long
+        if variant is None:
+            file_url = f"https://github.com/ramonaoptics/teensy-to-any/releases/download/{version}/firmware_{mcu.lower()}.hex"  # noqa # pylint: disable=line-too-long
+        else:
+            file_url = f"https://github.com/ramonaoptics/teensy-to-any/releases/download/{version}/firmware_{mcu.lower()}_{variant}.hex"  # noqa # pylint: disable=line-too-long
```

Comment:
```suggestion
        release_url = f"https://github.com/ramonaoptics/teensy-to-any/releases/download/{version}/"
        if variant is None:
            file_url = release_url + "firmware_{mcu.lower()}.hex"
        else:
            file_url = release_url + "firmware_{mcu.lower()}_{variant}.hex"
```

sometimes line too long is good warning!